File: msr/reference/zig/title-case.zig
const std = @import("std");

pub fn title_case(allocator: std.mem.Allocator, s: []const u8) ![]u8 {
    const out = try allocator.alloc(u8, s.len);
    var start = true;
    for (s, 0..) |c, i| {
        if (c == ' ') {
            out[i] = c;
            start = true;
        } else {
            out[i] = if (start) std.ascii.toUpper(c) else std.ascii.toLower(c);
            start = false;
        }
    }
    return out;
}

File: msr/reference/go/title-case.go
package main

import "strings"

func title_case(s string) string {
	words := strings.Split(s, " ")
	for i, w := range words {
		if w != "" {
			words[i] = strings.ToUpper(w[:1]) + strings.ToLower(w[1:])
		}
	}
	return strings.Join(words, " ")
}
